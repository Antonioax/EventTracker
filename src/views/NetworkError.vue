<template>
  <div class="text-center">
    <h1 class="text-5xl mb-6">Uh-Oh!</h1>
    <h3 class="text-lg">
      It looks like you're experiencing some network issues, <br>please take a breath and
      <a href="#" @click="$router.go(-1)" class="text-emerald-800">click here</a> to try again.
    </h3>
  </div>
</template>
