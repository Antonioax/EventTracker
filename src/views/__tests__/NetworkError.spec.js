import { mount } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";

import NetworkError from "@/views/NetworkError.vue";
import { createRouter, createWebHistory } from "vue-router";

describe("NetworkError.vue", () => {
  let wrapper;
  let router;

  beforeEach(() => {
    router = createRouter({
      history: createWebHistory(),
      routes: []
    });

    vi.spyOn(router, "go");

    wrapper = mount(NetworkError, {
      global: {
        plugins: [router]
      }
    });
  });

  it("should mount component", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should render text", () => {
    expect(wrapper.text()).toContain("It looks like you're experiencing some network issues");
  });

  it("should go back on link click", async () => {
    const link = wrapper.find("a");

    await link.trigger("click");

    expect(router.go).toHaveBeenCalledWith(-1);
  });
});
