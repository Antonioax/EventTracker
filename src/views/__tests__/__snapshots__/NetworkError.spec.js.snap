// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NetworkError.vue > should match snapshot 1`] = `
"<div class="text-center">
  <h1 class="text-5xl mb-6">Uh-Oh!</h1>
  <h3 class="text-lg"> It looks like you're experiencing some network issues, <br>please take a breath and <a href="#" class="text-emerald-800">click here</a> to try again. </h3>
</div>"
`;
